opts/hosts_test.go

```diff
@@ -8,19 +8,19 @@ import (
 
 func TestParseHost(t *testing.T) {
 	invalid := map[string]string{
-		"something with spaces": `parse "tcp://something with spaces": invalid character " " in host name`,
-		"://":                   `Invalid bind address format: ://`,
-		"unknown://":            `Invalid bind address format: unknown://`,
-		"tcp://:port":           `parse "tcp://:port": invalid port ":port" after host`,
-		"tcp://invalid:port":    `parse "tcp://invalid:port": invalid port ":port" after host`,
-		"tcp://:5555/":          `invalid bind address (:5555/): should not contain a path element`,
-		"tcp://:5555/p":         `invalid bind address (:5555/p): should not contain a path element`,
-		"tcp://0.0.0.0:5555/":   `invalid bind address (0.0.0.0:5555/): should not contain a path element`,
-		"tcp://0.0.0.0:5555/p":  `invalid bind address (0.0.0.0:5555/p): should not contain a path element`,
-		"tcp://[::1]:/":         `invalid bind address ([::1]:/): should not contain a path element`,
-		"tcp://[::1]:5555/":     `invalid bind address ([::1]:5555/): should not contain a path element`,
-		"tcp://[::1]:5555/p":    `invalid bind address ([::1]:5555/p): should not contain a path element`,
-		" tcp://:5555/path ":    `invalid bind address (:5555/path): should not contain a path element`,
+		"something with spaces": `invalid bind address (something with spaces): parse "tcp://something with spaces": invalid character " " in host name`,
+		"://":                   `invalid bind address (://): unsupported proto ''`,
+		"unknown://":            `invalid bind address (unknown://): unsupported proto 'unknown'`,
+		"tcp://:port":           `invalid bind address (tcp://:port): parse "tcp://:port": invalid port ":port" after host`,
+		"tcp://invalid:port":    `invalid bind address (tcp://invalid:port): parse "tcp://invalid:port": invalid port ":port" after host`,
+		"tcp://:5555/":          `invalid bind address (tcp://:5555/): should not contain a path element`,
```

Comment:
With this change, the actual input is included, not a "partial" representation of what the user passed. Previously the error would mention a valid that I (as a user) didn't pass, which made it rather confusing.